Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp

```diff
@@ -789,8 +789,22 @@ IGNORE_GCC_WARNINGS_END
 
         emitNewGenerator(m_generatorRegister);
         bool isInternalPromise = false;
+#if USE(BUN_JSC_ADDITIONS)
+        if (m_isBuiltinFunction) {
+            const auto isDefaultPromiseForBun = [&]() -> bool {
+                if (auto* provider = m_scopeNode->source().provider()) {
+                    const String& sourceURL = provider->sourceURL();
+                    if (!sourceURL.isNull() && !sourceURL.isEmpty())
```

Comment:
```suggestion
                    return !sourceURL.isEmpty();
```

Reply:
If there are no use cases where we want to use `JSInternalPromise` in Bun's built-ins, sounds good.